### PR TITLE
Including code review tools for CI

### PR DESCRIPTION
**Pull request by carmat88**
## Change content and motivation
Including and improving CI with some code review tools such as:

- `shellcheck`
- `python -mjson.tool`

and command feature such as `terraform fmt`

## GitHub cross-links 

**Fixes:** 

**Docs**:

### CODE REVIEW

**Comment by carmat88** [comment]
@mcapuccini Code review tools have been added to the CI for the image building. Please let me know if you have any comments. Cheers.

File: .travis.yml [hunk 1]
```diff
@@ -20,63 +27,71 @@ before_install:
...
+  - >
+      travis_retry curl
+      https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip
+      > /tmp/terraform.zip
+  - unzip /tmp/terraform.zip -d /tmp/
+  - /tmp/terraform fmt bin/gce_main.tf
+  - git diff --exit-code # this will fail if terraform changed something
+  
+  # Check Json
+  - python -mjson.tool < build-"$HOST_CLOUD".json
```
Comment (mcapuccini) [inline .travis.yml]: This always return 0. You need to do the trick with `git diff --exit-code` like above:

*untested*
```
echo "$(python -mjson.tool < build-aws.json)" > build-aws.json 
git diff --exit-code
```

File: .travis.yml [hunk 2]
```diff
@@ -20,63 +27,71 @@ before_install:
...
+  
+  # check Shell
+  - shellcheck -x $(find . -type f -name "*.sh")
```
Comment (mcapuccini) [inline .travis.yml]: you can remove `-x`

**Review by mcapuccini — CHANGES_REQUESTED** [review]
there is a problem with the json formatting, the rest looks good!

File: .travis.yml (hunk 1 above)
Comment (carmat88) [reply .travis.yml]: Cool, many thanks. I just tested them locally. First time returned 1, then (after the commit of course) returned 0. Will push now changes.

File: .travis.yml (hunk 2 above)
Comment (carmat88) [reply .travis.yml]: Ok, done. Will push soon change.

**Review by mcapuccini — APPROVED** [review]

**Pull request merged**